DragonSlayer.h

```diff
@@ -1,8 +1,15 @@
 #pragma once
 
+#include "Character.h"
+#include "AttackItem.h"
+
 struct DragonSlayer : Character
 {
+    DragonSlayer(const std::string n, int hp, int a);
     void attack(Character& other) override;
+    const std::string& getName() override;
+    std::string getStats() override;
 private:
     const std::string name;
-};
\ No newline at end of file
+    AttackItem at;
```

Comment:
As you can only use items once, this should be a `std::unique_ptr<AttackItem>` that resets after use. 

- [ ] change AttackItem to `unique_ptr<AttackItem>`